validations/src/tests.rs

```diff
@@ -6912,3 +6935,119 @@ fn validate_commit_transactions_included_in_utxo_diff() {
+        let vt_tx = VTTransaction::new(vt_body, vec![vts]);
+
+        assert_eq!(vt_tx.weight(), 493);
+
+        (
+            BlockTransactions {
+                value_transfer_txns: vec![vt_tx],
+                ..BlockTransactions::default()
+            },
+            1_000_000 - 10,
+        )
+    };
+    let x = test_blocks_with_limits(vec![t0], 492, 0);
+    assert_eq!(
+        x.unwrap_err().downcast::<BlockError>().unwrap(),
+        BlockError::ValueTransferWeightOverflow {
+            weight: 493,
+            max_weight: 492,
+        },
+    );
+}
+
+#[test]
+fn validate_vt_weight_valid() {
+    let t0 = {
+        let vto0 = ValueTransferOutput {
+            time_lock: 0,
+            pkh: Default::default(),
+            value: 10,
+        };
+        let output1_pointer = MILLION_TX_OUTPUT.parse().unwrap();
+        let vt_body = VTTransactionBody::new(vec![Input::new(output1_pointer)], vec![vto0]);
+        let vts = sign_tx(PRIV_KEY_1, &vt_body);
+        let vt_tx = VTTransaction::new(vt_body, vec![vts]);
+
+        assert_eq!(vt_tx.weight(), 493);
+
+        (
+            BlockTransactions {
+                value_transfer_txns: vec![vt_tx],
+                ..BlockTransactions::default()
+            },
+            1_000_000 - 10,
+        )
+    };
+    let x = test_blocks_with_limits(vec![t0], 493, 0);
+    x.unwrap();
+}
+
+#[test]
+fn validate_dr_weight_overflow() {
+    let t0 = {
+        let output1_pointer = MILLION_TX_OUTPUT.parse().unwrap();
+        let dro = example_data_request_output(2, 1, 0);
+        let dr_value = dro.checked_total_value().unwrap();
+
+        let dr_body = DRTransactionBody::new(vec![Input::new(output1_pointer)], vec![], dro);
+        let drs = sign_tx(PRIV_KEY_1, &dr_body);
+        let dr_tx = DRTransaction::new(dr_body, vec![drs]);
+
+        assert_eq!(dr_tx.weight(), 1558);
+
+        (
+            BlockTransactions {
+                data_request_txns: vec![dr_tx],
+                ..BlockTransactions::default()
+            },
+            1_000_000 - dr_value,
+        )
+    };
+    let x = test_blocks_with_limits(vec![t0], 0, 1558 - 1);
+    assert_eq!(
+        x.unwrap_err().downcast::<BlockError>().unwrap(),
+        BlockError::DataRequestWeightOverflow {
+            weight: 1558,
+            max_weight: 1558 - 1,
+        },
+    );
+}
+
+#[test]
+fn validate_dr_weight_valid() {
+    let t0 = {
+        let output1_pointer = MILLION_TX_OUTPUT.parse().unwrap();
+        let dro = example_data_request_output(2, 1, 0);
+        let dr_value = dro.checked_total_value().unwrap();
+
+        let dr_body = DRTransactionBody::new(vec![Input::new(output1_pointer)], vec![], dro);
+        let drs = sign_tx(PRIV_KEY_1, &dr_body);
+        let dr_tx = DRTransaction::new(dr_body, vec![drs]);
+
+        assert_eq!(dr_tx.weight(), 1558);
+
+        (
+            BlockTransactions {
+                data_request_txns: vec![dr_tx],
```

Comment:
Can we make a test ensuring that our calculations are correct? Checking for instance that  a DR with witneses > 125 does not fit in the block

Reply:
done!